proposals/stf-general-investment.md

```diff
@@ -0,0 +1,263 @@
+
+  Streamlining the contribution process should make it more attractive for newcomers to participate in development, and keep existing people around and motivated.
+  Growing the contributor community will make the ecosystem more robust against fluctuation.
+  We want to achieve this by making more systemic knowledge explicit and easier to discover, access, and manipulate, and thus ease onboarding of new contributors, and reduce contribution turnaround time.
+
+  - Tools for review automation (50k)
+  - Modularise the Nix configuration system (30k)
+  - Increase coverage and reduce duration of Nix integration testing (30k)
+  - Monitoring and data gathering for informed decision-making (30k)
+  - Professionalise the moderation team (50k)
+  - Organisation-wide declarative permission and notification management (60k)
+
+- Improve documentation (170k)
+
+  More focused, correct, up-to-date, and well-structured documentation will make the system more learnable.
+  This will over time reduce the number and size of blind spots in components that may otherwise end up unmaintained, a problem we are observing for many years now.
+  It should also tangibly reduce maintainer workload by making users and contributors more self-sufficient.
+
+  - Improve and speed up documentation rendering infrastructure (40k)
+  - Rework the Nixpkgs manual (60k)
+  - [Design and implement an information architecture](https://discourse.nixos.org/t/zurich-23-11-zhf-hackathon-and-ux-workshop-report/37848#ux-workshop-5) (50k)
+  - Build documentation testing infrastructure (20k)
+
+- Reduce ongoing costs for continuous integration and distribution (180k)
+
+  Infrastructure expenses are a signfificant risk for the ecosystem, because they are almost completely sponsored by companies who may end their support any time.
+  Reducing these ongoing costs will make it easier to migrate between providers or even self-host.
+  Improving CI performance will also contribute to faster workflows, and increase responsiveness to security incidents.
+  This will also impact all downstream users and significantly reduce their storage and energy requirements for the forseeable future.
+
+  - Deduplicate the cache storage (40k)
+  - Improve evaluation performance (100k)
+    - parallel evaluation
+    - evaluation cache
+    - memory layout optimisations
+    - minimal NixOS modules list
+  - Rewrite the Hydra front-end (40k)
+
+- Strengthen supply-chain and end-user security (300k)
+
+  At the scale of Nixpkgs, we have many attack surfaces that are currently not being taken care of appropriately.
+  The proposed measures will make common interactions with the Nix ecosystem substantially more secure by default.
+
+  - Rootless Nix daemon (50k)
+  - Roll out secure boot capabilities (30k)
+  - Secure ofBorg checks (55k)
+  - Scriptless system activation (30k)
+  - Roll out full-source bootstrapping (50k)
+  - Make vulnerability tracking a first-class use case (70k)
```

Comment:
I think a security advisory system like [Gentoo Linux Security Advisories (GLSA)][GLSA] ([more info][wiki]) is of great benefit to end users.  Is it included in this task?

[GLSA]: https://security.gentoo.org/glsa
[wiki]: https://wiki.gentoo.org/wiki/GLSA

Reply:
Was started at https://github.com/Nix-Security-WG/nix-security-tracker